src/material/list/list.ts

```diff
@@ -58,6 +58,7 @@ export class MatList extends MatListBase {}
     '[class.mdc-list-item--with-leading-icon]': '_icons.length !== 0',
     '[class.mdc-list-item--with-trailing-meta]': '_meta.length !== 0',
     '[class._mat-animation-noopable]': '_noopAnimations',
+    '[attr.aria-current]': '_ssrSafeIsCurrentPage() ? "page" : null',
```

Comment:
What makes this "SSR-safe"? An expression like `_activated ? "page" : null` is safe as well.

Reply:
It's SSR safe because it confirms that the hostElement is an anchor by checking the `nodeName` property. This is SSR safe but it wouldn't be safe to do `instanceof HTMLAnchorElement`.

I'm okay with inlining `_ssrSafeIsCurrentPage` if that's easier.

Reply:
Shouldn't we add it to non-anchors as well?

Reply:
It's only appropriate to put `aria-current="page"` on an anchor (or maybe something else with role="link"). Not applicable to other elements here because they're not supposed to be links. From [MDN aria-current](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-current):

> [`page`] Represents the current page within a set of pages such as the link to the current document in a breadcrumb.

Does that sound okay to you?

Reply:
I was under the impression that the navigation list was meant to be used exclusively with links.

Reply:
Well, maybe the commit message is a bit misleading because it mentions `<nav-list/>`.

Here's what I'm trying to accomplish. From the documentation for `activated` Input on mat-chip:

```
  /** Indicates whether an item in a `<mat-nav-list>` is the currently active page. */
```

In the situation where there is an active link (e.g. `a matListItem activated href="foo">...`). The activated state is only communicated visually by the activated CSS visual state. For correct a11y, the list item aslo needs communicated that it's activated to the a11y tree.

`aria-current="page"` is the best solution for this IMHO because by definition it indicates that the link is the active page in the browser.

Technically, a <mat-nav-list> would be the most appropriate container for an anchor tag list item, but it can be used inside a normal mat-list. The best I could do to detect that we needed to put aria-current="page" is if the list item is an anchor tag and activated is true. I didn't think it was necessary to also check that the parent is also a mat-nav-list. If mat-list-item doesn't need to check that it's parent is a mat-nav-list, it wouldn't be hard for me to change it to do that.

WDYT?

Reply:
thanks, I can also add a unit test for this, since it's sufficiently complicated

Reply:
Okay, I'll add a unit test to make sure that aria-current is added to list-items that are anchor tags, but I won't implement any checks that the parent is a nav-list (and not a mat-list)